prac_07/programming_language.py

```diff
@@ -0,0 +1,44 @@
+"""
+CP1404/CP5632 Practical - Suggested Solution
+Programming Language class with tests.
+"""
+
+
+class ProgrammingLanguage:
+    """Represent information about a programming language."""
+
+    def __init__(self, name, typing, reflection, year, pointer_arithmetic):
+        """Construct a ProgrammingLanguage from the given values."""
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+        self.pointer_arithmetic = pointer_arithmetic
+
+    def __repr__(self):
+        """Return string representation of a ProgrammingLanguage."""
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, " \
+               f"Pointer Arithmetic={self.pointer_arithmetic}, First appeared in {self.year}"
+
+    def is_dynamic(self):
+        """Determine if language is dynamically typed."""
+        return self.typing == "Dynamic"
+
+
+def run_tests():
+    """Run simple tests/demos on ProgrammingLanguage class."""
```

Comment:
Consider updating these tests to use the pointer_arithmetic value.
Could also be useful to make a test specifically for the new value (for example, display all programming names where pointer_arithmetic is true).

Reply:
Good spoting. I will add some testing for this section to prove that it works correctly.